source/common/http/http2/codec_impl.cc

```diff
@@ -894,10 +923,19 @@ void ConnectionImpl::StreamImpl::resetStreamWorker(StreamResetReason reason) {
     return;
   }
   if (codec_callbacks_) {
-    codec_callbacks_->onCodecLowLevelReset();
+    // TODO(wbpcode): this ensure that onCodecLowLevelReset is only called once. But
+    // we should replace this with a better design later.
+    // See https://github.com/envoyproxy/envoy/issues/42264 for why we need this.
+    if (!codec_low_level_reset_is_called_) {
+      codec_low_level_reset_is_called_ = true;
+      codec_callbacks_->onCodecLowLevelReset();
```

Comment:
thanks for the guard, I think we we should also change the log https://github.com/envoyproxy/envoy/blob/156480162c0007fc477f39d96e8610ab220d2786/source/common/http/conn_manager_impl.cc#L2103 here.